src/lib.rs

```diff
@@ -1093,6 +1100,46 @@ impl<K, V, S> Drop for FlurryHashMap<K, V, S> {
     }
 }
 
+impl<K, V, S> Extend<(K, V)> for &FlurryHashMap<K, V, S>
+where
+    K: Sync + Send + Clone + Hash + Eq,
+    V: Sync + Send,
+    S: BuildHasher,
+{
+    // TODO: Implement Java's `tryPresize` method to pre-allocate space for
+    // the incoming entries
+    fn extend<T: IntoIterator<Item = (K, V)>>(&mut self, iter: T) {
+        let guard = crossbeam::epoch::pin();
+
+        for (key, value) in iter {
+            (*self).put(key, value, false, &guard);
+        }
+    }
+}
+
+impl<K, V> FromIterator<(K, V)> for FlurryHashMap<K, V, RandomState>
+where
+    K: Sync + Send + Clone + Hash + Eq,
+    V: Sync + Send,
+{
+    fn from_iter<T: IntoIterator<Item = (K, V)>>(iter: T) -> Self {
+        let mut it = iter.into_iter();
+
+        let map = if let Some((key, value)) = it.next() {
+            let guard = crossbeam::epoch::pin();
+            let (lower, _) = it.size_hint();
+            let map = Self::with_capacity(lower.saturating_add(1));
+            map.put(key, value, false, &guard);
+            map
+        } else {
+            Self::new()
```

Comment:
I think this can be a `return Self::new()`, since we know here that the `.extend` isn't necessary (since the iterator has ended).

Reply:
Yep, you're right.